### PR TITLE
NAS-122206 / 23.10 / Investigate `dataset does not exist` when selecting source in Replication Wizard

### PR DESCRIPTION
**Pull request by AlexKarpov98**
result:
Screenshot 2023-06-02 at 14 10 58

### CODE REVIEW

**Comment by undsoft** [comment]
@AlexKarpov98 Have you looked into why there are non-existing dataset available in ix-explorer here?

**Comment by AlexKarpov98** [comment]
@undsoft - it's the data which comes from the middleware, not sure why it grabs non datasets.
`filesystem.listdir`

with such parameters:
`/mnt/tank [] {order_by: Array(1), limit: 1000}`

I see that middleware sends `file.is_mountpoint` property, if we filter by it - we only see available datasets.

**Comment by AlexKarpov98** [comment]
@undsoft - make sense!

File: src/app/services/filesystem.service.ts [hunk 1]
```diff
@@ -40,11 +40,10 @@ export class FilesystemService {
         map((files) => {
           const children: ExplorerNodeData[] = [];
           files.forEach((file) => {
-            if (file.type === FileType.Symlink || !file.hasOwnProperty('name')) {
-              return;
-            }
-
-            if (!options.showHiddenFiles && file.name.startsWith('.')) {
+            if (
```
Comment (undsoft) [inline src/app/services/filesystem.service.ts:43]: Previous version was more readable to me.

**Review by undsoft — CHANGES_REQUESTED** [review]
I get errors when I try to use it.

https://github.com/truenas/webui/assets/1481270/cfa89c43-ac9a-4845-acad-0643e81964f2

**Comment by AlexKarpov98** [comment]
@undsoft - are you sure that you are checking on the correct branch with latest updates? (I tested it and it works fine)

**Pull request merged**